scipy/integrate/_quadrature.py

```diff
@@ -1035,3 +1039,242 @@ def newton_cotes(rn, equal=0):
+    error of the mean :math:`s` of these estimates may be used to generate
+    confidence intervals using the t distribution with ``n_estimates - 1``
+    degrees of freedom.
+
+
+    Examples
+    --------
+    QMC quadrature is particularly useful for computing integrals in higher
+    dimensions; an example integrand is the probability density function
+    of a multivariate normal distribution.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> dim = 8
+    >>> mean = np.zeros(dim)
+    >>> cov = np.eye(dim)
+    >>> def func(*z):
+    ...     z = np.asarray(z).T
+    ...     return stats.multivariate_normal.pdf(z, mean, cov)
+
+    To compute the integral over the unit hypercube:
+
+    >>> from scipy.integrate import qmc_quad
+    >>> lb = np.zeros(dim)
+    >>> ub = np.ones(dim)
+    >>> ranges = np.asarray([lb, ub]).T
+    >>> rng = np.random.default_rng()
+    >>> qrng = stats.qmc.Halton(d=dim, seed=rng)
+    >>> n_estimates = 8
+    >>> res = qmc_quad(func, ranges, n_estimates=n_estimates, qrng=qrng)
+    >>> res.integral, res.standard_error
+    (0.00018443157359780213, 8.95876227122206e-08)
+
+    A two-sided, 99% confidence interval for the integral may be estimated
+    as:
+
+    >>> t = stats.t(df=n_estimates-1, loc=res.integral,
+    ...             scale=res.standard_error)
+    >>> t.interval(0.99)
+    (0.00018411806320847138, 0.0001847450839871329)
+
+    Indeed, the value reported by `scipy.stats.multivariate_normal` is
+    within this range.
+
+    >>> stats.multivariate_normal.cdf(ub, mean, cov, lower_limit=lb)
+    0.00018430867675187443  # may vary
+
+    """
+    args = _qmc_quad_iv(func, ranges, n_points, n_estimates, qrng, log, args)
+    func, ranges, n_points, n_estimates, qrng, rng, log, args, stats = args
+
+    lb, ub = ranges.T
+
+    # The sign of the integral depends on the order of the limits. Fix this by
+    # ensuring that lower bounds are indeed lower and setting sign of resulting
+    # integral manually
+    i_swap = ub < lb
+    sign = (-1)**(i_swap.sum(axis=-1))  # odd # of swaps -> negative
+    lb[i_swap], ub[i_swap] = ub[i_swap], lb[i_swap]
+
+    A = np.prod(ub - lb)
+    dA = A / n_points
+
+    estimates = np.zeros(n_estimates)
+    for i in range(n_estimates):
+        # Generate integral estimate
+        sample = qrng.random(n_points)
+        x = stats.qmc.scale(sample, lb, ub)
+        integrands = func(*x.T, *args)
+        if log:
+            estimate = logsumexp(integrands) + np.log(dA)
+        else:
+            estimate = np.sum(integrands * dA)
+        estimates[i] = estimate
+
+        # Get a new, independently-scrambled QRNG for next time
+        if hasattr(qrng._init, 'scramble'):
+            qrng._init['scramble'] = True
+        # Halton needs us to either burn RNG samples or spawn a new Generator
+        rng = _rng_spawn(qrng.rng, 1)[0]
+        qrng = type(qrng)(seed=rng, **qrng._init)
```

Comment:
Since this application is one of the main reasons for scrambling (per the documentation), we might want to think about a more convenient API on `QMCEngine` that would help do this more naturally. Not necessarily in this PR, but this PR would be a very good vehicle for it, especially if it had implications for the interface of `qmc_quad` (for example, maybe the way to do this isn't to pass in a `QMCEngine` instance at all but just the class).

Reply:
> maybe the way to do this isn't to pass in a QMCEngine instance at all but just the class

It was a choice to make the user pass in an instance. If we accept just the class, we also need to add a way for user to configure it with options like `optimization`, `seed`, etc. I would prefer to avoid that. 

---

Along the same lines, I would also like to remove the `args` input. I included it for consistency with `nquad`, but that's inconsistent with itself - why accept `args` but not `kwargs`? The user should just wrap their own function so that it doesn't need additional arguments. This reduces clutter in the documentation and avoids concern about what to do when `args` is not a tuple.

Reply:
Yes, we would prefer to avoid passing the class and arguments, but after examining all of the options for implementing this pattern, it might be the case that that's better than the alternatives. That's why this PR might be the best vehicle for discussing the API changes to `QMCEngine` that would facilitate this pattern rather than committing to an interface for `qmc_quad()` now and redoing `QMCEngine` later.

Reply:
It seems to me that the `qrng` argument could be used in the future to accept a class, rather than an instance of a class, if we were to decide to change something.